### ISSUE TITLE
How to traverse images and graphs in dataloader simultaneously?

### ISSUE THREAD

**Issue by lichen0411** [rusty1s/pytorch_geometric#1678]
I have datasets that contains image data and its corresponding graph data. Then I want to use the images and its corresponding graphs in a batch to train the model, but the two shuffles results different permutations.

- trainloader_image = torch.utils.data.DataLoader(trainset_image, batch_size=128, shuffle=True)
- trainloader_graph = DenseDataLoader(trainset_graph, batch_size=128, shuffle=True)

When I try to traverse the two dataloaders simultaneously using following codes:
- for batch_idx, (inputs, targets), data in enumerate(trainloader_image,trainloader_graph):
The error occurs that ‘DenseDataLoader’ object cannot be interpreted as an interger.
So how can I handle the problem? Thanks.

**Comment by rusty1s** [comment]
I think for this use-case it's best to create a "wrapper dataset", *e.g.*:
```python
data_list = [(x, y) for x, y in zip(trainset_image, trainset_graph)]
loader = torch_geometric.data.DataLoader(data_list, batch_size=128, shuffle=True)
```

**Comment by lichen0411** [comment]
Thanks for your answer. I have tried this implementation way, however when fetch the data x for trainset_image in loader, the error occurs that "‘tuple’ object has no attribute 'keys'". It's mainly due to that the data form is different for images and graphs and therefore the dataloader used is different(torch.utils.data.DataLoader, torch_geometric.data.DataLoader). The image data cannot be fetched directly using torch_geometric.data.DataLoader.

**Comment by rusty1s** [comment]
Which version of PyG are you using? This should have been fixed in one of the later releases.

**Comment by lichen0411** [comment]
I'm using the PyG version 1.6.1 now.

**Comment by lichen0411** [comment]
Thanks a lot! I've solved the problem now. I use torch_geometric.datasets.data.image and torch_geometric.datasets.slices['image'] to conclude the image data in my graph data, then I use the torch_geometric.data.DataLoader to fetch image data and graph data in a batch to train the model.

**Comment by rusty1s** [comment]
Cool that you have fixed your issues. However, I cannot reproduce your error mentioned earlier. For example, the following code runs without any problems for me:
```python
import torch
from torch_geometric.datasets import TUDataset
from torch_geometric.data import DataLoader

dataset = TUDataset('/tmp/TUDataset', 'MUTAG')

data_list = [(torch.randn(28, 28), data) for data in dataset]
loader = DataLoader(data_list, batch_size=32, shuffle=True)

for (img, data) in loader:
    print(img.shape)
    print(data)
```

**Comment by lichen0411** [comment]
Well, I've tried your code. It runs without any problem. The error occurs before is mainly due to that the img is stored in tuple format not the tensor format. Change the img to tensor format may solve the problem and I'll verify that. Thanks!
